src/days-2-4-adv/adv.py

```diff
@@ -49,3 +50,39 @@
 # Print an error message if the movement isn't allowed.
 #
 # If the user enters "q", quit the game.
+
+while True:
+    error = "You Shall Not Pass"
+
+    print(f"{player.inRoom.name}")
+    print(f"{player.inRoom.description}")
+
+    controller = input("Where do we go from here?")
+
+    if controller == "q":
+        print("Let's Leave!")
+        break
+    
+    elif controller == 'n':
+        if hasattr(player.inRoom, 'n_to'):
+            player.inRoom(player.inRoom.n_to)
+        else:
+            print(error)
+    
+    elif controller == 's':
+        if hasattr(player.inRoom, 's_to'):
+            player.inRoom(player.inRoom.s_to)
+        else:
+            print(error)
+    
+    elif controller == 'e':
+        if hasattr(player.inRoom, 'e_to'):
+            player.inRoom(player.inRoom.e_to)
+        else:
+            print(error)
+    
+    elif controller == 'w':
+        if hasattr(player.inRoom, 'w_to'):
+            player.inRoom(player.inRoom.w_to)
+        else:
+            print(error)
```

Comment:
One thing I forgot to mention: What will your program do if it receives an input you haven't coded for yet? What do you want it to do?